### DISCUSSION TITLE
Is it possible to use custom routes and retain the query string?

### DISCUSSION THREAD

**Discussion by soupaJ** [vercel/next.js#10698] [Help]
I'm testing the experimental redirects feature with a few routes to replace an older redirect component that we use. During my initial test, I noticed that the query string isn't carried through to the destination. This causes my redirects to not work as intended if the user came from an older link which had specific filters as well as stripping query parameters consumed by our analytics like campaign, source, etc. Is there a way to create a source and destination pattern which will carry all query params through without explicitly defining each query param in the pattern?

**Comment by ijjk** [comment]
Hi, currently passing all query params through from the request in a redirect isn't directly supported and we only apply query values you specify on the destination directly to prevent accidentally including values that aren't intended to be passed.

This feature is still experimental and this may change although you could use a rewrite to an API route that set's the redirect header passing all of the query values from the request to achieve this behavior currently if desired

**Comment by wtakayama-chwy** [comment]
I'm not sure if it can help you, but recently I've come to some similar issue and I implemented the following code:

TIP: this is a function that we use for our product page which is created dynamically according the data from our db

![image](https://user-images.githubusercontent.com/60292132/105856389-f6e0f280-5fc7-11eb-8e36-15c4b9400ec1.png)
